shared/src/main/scala/mlscript/codegen/Scope.scala

```diff
@@ -253,13 +259,53 @@ class Scope(name: Str, enclosing: Opt[Scope]) {
       base: Type,
       methods: Ls[MethodDef[Left[Term, Type]]],
       ctor: Ls[Statement],
-      superParameters: Ls[Term]
+      superParameters: Ls[Term],
+      nested: Ls[NuTypeDef],
+      isNested: Bool
+  ): ModuleSymbol = {
+    val symbol = ModuleSymbol(lexicalName, params.sorted, base, methods, ctor, superParameters, nested, isNested)
+    register(symbol)
+    symbol
+  }
```

Comment:
What's the point of keeping all these `declareX` methods? They seem to be pure redundant noise. Just create the value inline, as in `ModuleSymbol(...).tap(register)`.

Reply:
Did you mean remove all `declareX` methods and create a helper method `tap` in the symbol classes for registering?

Reply:
`tap` is a standard Scala method. And yes I don't see why these `declareX` methods that only wrap a simple object creation are needed. They're pure noise.

Reply:
Removed.